### ISSUE TITLE
AC is not calculated for the samples in the VCF file with bcftools +aggregate

### ISSUE THREAD

**Issue by tommycarstensen** [samtools/bcftools#563]
I have several AC annotations in my VCF file. For example `AC_Uganda`, `AC_Africa`, `AC`. `AC_Uganda` is a subset of `AC_Africa`, but instead of `AC` being a summed allele count for the samples in the VCF file it is a sum across all `AC` values; i.e. `AC_Uganda` and `AC_Africa` are summed. This is definitely not the behaviour I expect as a user.

Here some real values from a VCF file that were calculcated with `bcftools +aggregate -Ou -- -S samples.aggregate.tsv`:
```
AC=13984
AC_KHV=169
AC_supEAS=845
AC_GIH=196
AC_supSAS=921
AC_JPT=180
AC_STU=193
AC_GWD=179
AC_supAFR=3620
AC_Zulu=175
AC_Egypt=186
AC_Oromo=40
AC_Khoesan=55
AC_PUR=192
AC_supAMR=618
AC_CHS=179
AC_PEL=153
AC_Uganda=1990
AC_ACB=163
AC_YRI=176
AC_CLM=163
AC_LWK=152
AC_IBS=212
AC_supEUR=988
AC_FIN=197
AC_ITU=194
AC_BEB=165
AC_MXL=110
AC_PJL=173
AC_MSL=139
AC_CEU=192
AC_ESN=150
AC_CDX=147
AC_TSI=212
AC_Somali=36
AC_CHB=170
AC_ASW=85
AC_GBR=175
AC_Amhara=35
AC_Wolayta=29
AC_Gumuz=30
```

**Comment by pd3** [comment]
Hi Tommy,
thank you for the bug report. It is a known bug, but since the tool is not present any more, a fix was never made available. The functionality was moved from the unofficial `aggregate` plugin into the `fill-tags` plugin recently, and revamped just yesterday (c799c5e). Could you please try that instead?

**Comment by tommycarstensen** [comment]
Thanks Petr! I did a `git pull` and `make clean`. I suspect `fill-tags` will work just fine. However, with this command:

```
| $bcftools +fill-tags -Ou -- -S samples.aggregate.tsv \
```

I get this error:
```
Error: subset called for sample that does not exist in header: "HG02029	KHV,supEAS". Use "--force-samples" to ignore this error.
```

When I add `--force-samples`, then I get this error:
```
fill-tags: unrecognized option '--force-samples'
```

Could you make `fill-tags` accept `--force-samples`? Thanks!

In the meantime I trust that `fill-tags` would have returned the correct values. I'll keep an eye on it, when I have to use it next. Thanks.

**Comment by pd3** [comment]
You don't want to use the `--force-samples` option here, the sample file is not being parsed correctly from some reason. I suspect this is not the latest version of the plugin. When you run `bcftools +fill-tags -h`, do you see the `HWE` annotation advertised? If you don't it's not the latest version of the plugin. If you do, can you point me to your files to reproduce the error?

**Comment by tommycarstensen** [comment]
Petr, it does indeed have the HWE option:
```
About: Set INFO tags AF, AC, AC_Hemi, AC_Hom, AC_Het, AN, HWE, MAF, NS.
```
```
   bcftools +fill-tags in.bcf -Ob -o out.bcf -- -S sample-group.txt -t HWE
```

Here is the tab separated samples file and its contents:
```
samples=/lustre/scratch115/projects/adrp1/users/tc9/SHAPEIT/samples.aggregate.tsv
grep -C1 HG02029 $samples
HG02029	KHV,supEAS
NA20897	GIH,supSAS
```

Here is one file, which can be used as input:
```
vcf=/lustre/scratch115/projects/adrp1/users/tc9/SHAPEIT/out_hapssample2vcf/20.vcf.gz
```

When receiving a binary piped input it seems to fail:
```
$bcftools annotate \
 -a $AR \
 -c QUAL,FORMAT/AD,FORMAT/DP,INFO,^INFO/AN,^INFO/AC,^INFO/AF,^FORMAT/GT,^FORMAT/GQ,^INFO/VariantType,^INFO/Dels \
 $(if [ $mask != "no_mask" ]; then echo " -R 20141020.$mask.whole_genome.bed.gz"; fi) \
 -Ou \
 $vcf \
| $bcftools annotate -a $VRtsv -h header.VR.txt -c CHROM,POS,INFO/VQSLOD -Ou \
| $bcftools annotate -a $VEPtsv -h header.VEP.txt -c CHROM,POS,REF,ALT,INFO/CSQ -Ou \
| $bcftools view $(if [ $related != "without_related" ]; then echo " -S ^../analysis/IBD/samples.nonfounders.txt"; fi) -Ou \
| $bcftools view --min-ac $minAC -T ^culprits.90.tsv.gz -Ou \
| $bcftools +fill-tags -Ou -- -S samples.aggregate.tsv \
| $bcftools view -Oz -o $out
```

As per this shell wrapper script:
/lustre/scratch115/projects/adrp1/users/tc9/SHAPEIT/annotate.sh

I'll just make sure that the samples in the VCF file and samples file correspond to each other. Thanks.

**Comment by tommycarstensen** [comment]
Petr, I've tried to regenerate this error with a simpler example, but I'm unable to. I tried piping the upstream output prior to `+fill-tags` to a new VCF file and then stream `bcftools view -Ou` to `+fill-tags`, but that doesn't reproduce the error. I'm giving up on this, but I'll let you know, if I'm able to reproduce the error with a simpler example in the future. Feel free to close it for now. Thanks and sorry for having wasted your time.
